src/components/new-safe/steps/Step3/index.tsx

```diff
@@ -0,0 +1,157 @@
+import type { NamedAddress } from '@/components/create-safe/types'
+import { useCurrentChain } from '@/hooks/useChains'
+import useGasPrice from '@/hooks/useGasPrice'
+import { useEstimateSafeCreationGas } from '@/components/create-safe/useEstimateSafeCreationGas'
+import { formatVisualAmount } from '@/utils/formatters'
+import type { StepRenderProps } from '@/components/new-safe/CardStepper/useCardStepper'
+import type { NewSafeFormData } from '@/components/new-safe/CreateSafe'
+import css from './styles.module.css'
+
+enum CreateSafeStep3Fields {
+  name = 'name',
+  owners = 'owners',
+  threshold = 'threshold',
+}
+
+type CreateSafeStep3Form = {
+  name: string
+  owners: NamedAddress[]
+  threshold: number
+  networkFee: string
+}
+
+const STEP_3_FORM_ID = 'create-safe-step-3-form'
+
+const ReviewRow = ({ name, value }: { name: string; value: ReactElement }) => {
+  return (
+    <>
+      <Grid item xs={3}>
+        <Typography>{name}</Typography>
+      </Grid>
+      <Grid item xs={9}>
+        {value}
+      </Grid>
+    </>
+  )
+}
+
+const CreateSafeStep3 = ({
+  onSubmit,
+  onBack,
+  data,
+}: Pick<StepRenderProps<NewSafeFormData>, 'onSubmit' | 'data' | 'onBack'>): ReactElement => {
+  const chain = useCurrentChain()
+  const { maxFeePerGas, maxPriorityFeePerGas } = useGasPrice()
+  const saltNonce = useMemo(() => Date.now(), [])
+
+  const safeParams = useMemo(() => {
+    return {
+      owners: data.owners.map((owner) => owner.address),
+      threshold: data.threshold,
+      saltNonce,
+    }
+  }, [data.owners, data.threshold, saltNonce])
+
+  const { gasLimit } = useEstimateSafeCreationGas(safeParams)
+
+  const totalFee =
+    gasLimit && maxFeePerGas && maxPriorityFeePerGas
+      ? formatVisualAmount(maxFeePerGas.add(maxPriorityFeePerGas).mul(gasLimit), chain?.nativeCurrency.decimals)
+      : '> 0.001'
+
+  const formMethods = useForm<CreateSafeStep3Form>({
+    mode: 'all',
+    defaultValues: {
+      [CreateSafeStep3Fields.name]: data.name,
+      [CreateSafeStep3Fields.owners]: data.owners,
+      [CreateSafeStep3Fields.threshold]: data.threshold,
+    },
+  })
+
+  const { handleSubmit, watch } = formMethods
+
+  const allFormData = watch()
+
+  const handleBack = () => {
+    onBack(allFormData)
+  }
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)} id={STEP_3_FORM_ID}>
+      <FormProvider {...formMethods}>
+        <Grid container spacing={3}>
+          <Grid item>
+            <Grid container spacing={3}>
+              <ReviewRow name="Network" value={<ChainIndicator chainId={chain?.chainId} inline />} />
+              <ReviewRow name="Name" value={<Typography>{data.name}</Typography>} />
+              <ReviewRow
+                name="Owners"
+                value={
+                  <Box className={css.ownersArray}>
+                    {data.owners.map((owner, index) => (
+                      <EthHashInfo address={owner.address} shortAddress={false} showPrefix={false} key={index} />
```

Comment:
This is missing the owner name/ens. I suggest to copy it from the other review component.